packages/meditrak-server/src/routes/importSurveys/ConfigImporter/processArithmeticConfig.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { ExpressionParser } from '@tupaia/expression-parser';
+import { splitStringOn, splitStringOnComma, translateExpression } from '../../utilities';
+
+export const processArithmeticConfig = async (models, config) => {
+  const { formula, defaultValues, valueTranslation, answerDisplayText } = config;
+  const expressionParser = new ExpressionParser();
+  const codes = expressionParser.getVariables(formula);
+
+  let translatedConfig = {
+    formula: await translateExpression(models, formula, codes),
+  };
+
+  if (defaultValues) {
+    translatedConfig = {
+      ...translatedConfig,
+      defaultValues: await translateDefaultValues(models, defaultValues),
+    };
+  }
+
+  if (valueTranslation) {
+    translatedConfig = {
+      ...translatedConfig,
+      valueTranslation: await translateValueTranslation(models, valueTranslation),
+    };
+  }
+
+  if (answerDisplayText) {
+    translatedConfig = {
+      ...translatedConfig,
+      answerDisplayText: await translateAnswerDisplayText(models, answerDisplayText, codes),
+    };
+  }
+
+  return translatedConfig;
+};
+
+/**
+ * defaultValues: SchFF00Section1Total:0,SchFF00Section1TotalObtainable:0
+ * will be converted to:
+ * defaultValues:{
+ *     5ed7558b61f76a6ba5003114:0,
+ *     5ec5bf8961f76a18ad00d4f9:0
+ * },
+ * @param {*} models
+ * @param {*} defaultValuesConfig
+ */
+const translateDefaultValues = async (models, defaultValuesConfig) => {
+  const defaultValues = splitStringOnComma(defaultValuesConfig);
+  const translatedDefaultValues = {};
+
+  for (const defaultValue of defaultValues) {
```

Comment:
Same comment before about fetching all questions at once from the db applies here and a couple of places below. Could be made into a util, using your same util code without the $ prepended.